File: backend/models/card.go
package models

import "time"

type StudentCard struct {
	StudentID        int64     `json:"student_id"`
	FirstName        string    `json:"first_name"`
	LastName         string    `json:"last_name"`
	EmployerName     *string   `json:"employer_name"` // Updated to pointer to handle NULL
	CheckInDateTime  time.Time `json:"check_in_date_time"`
	CheckOutDateTime time.Time `json:"check_out_date_time"`
	Emotion          string    `json:"emotion"`
	CheckInTime      string    `json:"check_in_time"`
	CheckOutTime     string    `json:"check_out_time"`
}

func (StudentCard) TableName() string {
	return "student_card"
}

type ErrorResponse struct {
	Error string `json:"error"`
}

File: web-app/src/types/card.ts
export interface Card {
  student_id: number;
  first_name: string;
  last_name: string;
  employer_name: string | null;
  check_in_date_time: string | null;
  check_out_date_time: string | null;
  check_in_time: string | null;
  check_out_time: string | null;
  emotion: string | null;
}
